services/create_event.go

```diff
@@ -0,0 +1,166 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/api/models"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/pkg/httpheader"
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/frain-dev/convoy/util"
+	"github.com/frain-dev/convoy/worker/task"
+	"github.com/oklog/ulid/v2"
+)
+
+var (
+	ErrInvalidEventDeliveryStatus  = errors.New("only successful events can be force resent")
+	ErrNoValidEndpointFound        = errors.New("no valid endpoint found")
+	ErrNoValidOwnerIDEndpointFound = errors.New("owner ID has no configured endpoints")
+	ErrInvalidEndpointID           = errors.New("please provide an endpoint ID")
+)
+
+type CreateEventService struct {
+	EndpointRepo datastore.EndpointRepository
+	Queue        queue.Queuer
+
+	NewMessage *models.CreateEvent
+	G          *datastore.Project
```

Comment:
```suggestion
	P          *datastore.Project
```
Sholdn't this be `P`?